### PR TITLE
 Made changes to report failures in .talismanrc as 'Warnings'

### PR DESCRIPTION
**Pull request by aaquibzama-tw**
This PR is intended to resolve the [Recursive Hex encoded issue](https://github.com/thoughtworks/talisman/issues/84)

Now, .talismanrc is reported only as a warning and not as an error

### CODE REVIEW

File: detector/filecontent_detector.go [hunk 1]
```diff
@@ -56,7 +58,11 @@ func fillResults(results []string, addition git_repo.Addition, result *Detection
 			log.WithFields(log.Fields{
 				"filePath": addition.Path,
 			}).Info(info)
-			result.Fail(addition.Path, fmt.Sprintf(output, res))
+			if strings.Contains(string(addition.Name), talismanrcfilename) {
+				result.Warn(addition.Path, fmt.Sprintf(output, res))
```
Comment (svishwanath-tw) [inline detector/filecontent_detector.go]: Why Contains ? Why not Equals ?

File: detector/detection_results.go [hunk 2]
```diff
@@ -36,6 +37,15 @@ func (r *DetectionResults) Fail(filePath git_repo.FilePath, message string) {
...
 
+func (r * DetectionResults) Warn(filePath git_repo.FilePath, message string) {
+	warnings, ok := r.warnings[filePath]
+	if !ok {
+		r.warnings[filePath] = []string{message}
+	} else {
+		r.warnings[filePath] = append(warnings, message)
+	}
+}
+
```
Comment (svishwanath-tw) [inline detector/detection_results.go]: There are some merge conflicts in this PR. Please address them.

File: detector/filecontent_detector.go (hunk 1 above)
Comment (sriharsha-y) [reply detector/filecontent_detector.go]: Rectified.

File: detector/detection_results.go (hunk 2 above)
Comment (sriharsha-y) [reply detector/detection_results.go]: Rectified.

**Pull request merged**